Sampling/roi_timeseries_masker.py

```diff
@@ -0,0 +1,43 @@
+import numpy as np
+from nibabel import load
+from nilearn.image import resample_img
+
+
+class ROITimeseriesMasker:
+    def __init__(self, path_to_roi_file, num_roi=376, background_label=0):
+        self.roi = load(path_to_roi_file)
+        self.num_roi = num_roi + 1  # add background label
+        self.background_label = background_label
+
+    def resample_roi(self, img):
+        resampled_roi = resample_img(self.roi,
+                                     target_affine=img.affine,
+                                     target_shape=img.shape[:3],
+                                     interpolation='nearest')
+        return resampled_roi
+
+    def mask_timeseries(self, path_to_img_file):
+        img = load(path_to_img_file)
+        resampled_roi = self.resample_roi(img)
+
+        img_data = img.get_fdata()
+        resampled_roi_data = resampled_roi.get_fdata()
+
+        roi_labels = list(np.unique(resampled_roi_data).astype(int))
+        roi_labels.remove(self.background_label)
+        roi_timeseries = {label: img_data[resampled_roi_data == label*1.0]
+                          for label in roi_labels}
+        return roi_timeseries
+
+
+if __name__ == '__main__':
+
+    path_to_img_file = '/Users/emmarocheteau/PycharmProjects/deep-attention-fMRI/Data/filtered_func_data_clean.nii.gz'
```

Comment:
I would suggest configuring the paths in PyCharm and refer to just Data rather than the full path.
```suggestion
    path_to_img_file = 'Data/filtered_func_data_clean.nii.gz'
```
```suggestion
    path_to_img_file = '/Users/emmarocheteau/PycharmProjects/deep-attention-fMRI/Data/filtered_func_data_clean.nii.gz'
```